### PR TITLE
step crl inspect

### PR DESCRIPTION
**Pull request by maraino**
### Description

This PR adds the new command `step crl inspect`. This command prints a crl in text, json, or PEM format.

### CODE REVIEW

File: command/crl/crl.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+)
+
+// init creates and registers the crl command
+func init() {
+	cmd := cli.Command{
+		Name:      "crl",
+		Usage:     "initialize and manage a certificate revocation list",
+		UsageText: "**step crl** <subcommand> [arguments] [global-flags] [subcommand-flags]",
+		Description: `**step crl** command group provides facilities to create, manage and inspect a
+certificate revocation list or CRL.
```
Comment (tashian) [inline command/crl/crl.go:15]:
```suggestion
certificate revocation list (CRL).
```

File: command/crl/inspect.go [hunk 2]
```diff
@@ -0,0 +1,522 @@
...
+	"go.step.sm/cli-utils/errs"
+	"go.step.sm/crypto/pemutil"
+	"go.step.sm/crypto/x509util"
+)
+
+func inspectCommand() cli.Command {
+	return cli.Command{
+		Name:      "inspect",
+		Action:    command.ActionFunc(inspectAction),
+		Usage:     "print certificate revocation list or CRL details in human readable format",
```
Comment (tashian) [inline command/crl/inspect.go:39]:
```suggestion
		Usage:     "print certificate revocation list (CRL) details in human-readable format",
```

File: command/crl/inspect.go [hunk 3]
```diff
@@ -0,0 +1,522 @@
...
+		UsageText: `**step crl inspect** <file|url>`,
+		Description: `**step crl inspect** prints the details of a certificate revocation list (CRL).
```
Comment (tashian) [inline command/crl/inspect.go:41]:
```suggestion
		Description: `**step crl inspect** validates and prints the details of a certificate revocation list (CRL). 
A CRL is considered valid if its signature is valid, the CA is not expired, and the next update time is in the future.
```

**Review by tashian — CHANGES_REQUESTED** [review]
lgtm with a couple of small suggestions.

File: command/crl/inspect.go [hunk 4]
```diff
@@ -44,7 +44,7 @@ A CRL is considered valid if its signature is valid, the CA is not expired, and
 ## POSITIONAL ARGUMENTS
 
 <file|url>
-:  The file or URL where the CRL is. If <**--from**> is passed it will inspect
+:  The file or URL where the CRL is. If <--from> is passed it will inspect
```
Comment (tashian) [inline command/crl/inspect.go:47]:
```suggestion
:  The file or URL where the CRL is. If <--from> is passed, it will inspect
```

**Review by tashian — APPROVED** [review]

File: command/crl/crl_extensions.go [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+
+func (e *Extension) MarshalJSON() ([]byte, error) {
+	return json.Marshal(e.json)
+}
+
+func (e *Extension) AddDetail(format string, args ...interface{}) {
+	e.Details = append(e.Details, fmt.Sprintf(format, args...))
+}
+
+func nexExtension(e pkix.Extension) Extension {
```
Comment (dopey) [inline command/crl/crl_extensions.go:99]: what is `nex`?

File: command/crl/inspect.go [hunk 6]
```diff
@@ -0,0 +1,523 @@
...
+## EXAMPLES
+
+Inspect a CRL:
+'''
+$ step crl inspect --insecure http://ca.example.com/crls/exampleca.crl
+'''
+
+Inspect and validate a CRL in a file:
+'''
+$ step crl inspect -ca ca.crt exampleca.crl
```
Comment (dopey) [inline command/crl/inspect.go:59]: `--ca`

File: command/crl/inspect.go [hunk 7]
```diff
@@ -0,0 +1,523 @@
...
+			return errors.Errorf("failed to get CRL distribution points from %s", crlFile)
+		}
+
+		crlFile = bundle[0].CRLDistributionPoints[0]
+		if len(bundle) > 1 {
+			caCerts = append(caCerts, bundle[1:]...)
+		}
+
+		if len(caCerts) == 0 && !ctx.Bool("insecure") {
+			println("foo")
```
Comment (dopey) [inline command/crl/inspect.go:198]: remove this line

**Review by dopey — CHANGES_REQUESTED** [review]
see comments

File: command/crl/inspect.go (hunk 7 above)
Comment (maraino) [reply command/crl/inspect.go]: Fixed with 2911198

File: command/crl/crl_extensions.go (hunk 5 above)
Comment (maraino) [reply command/crl/crl_extensions.go]: Fixed with 84c641c

File: command/crl/inspect.go (hunk 6 above)
Comment (maraino) [reply command/crl/inspect.go]: Fixed with 117f131

File: command/crl/inspect.go [hunk 8]
```diff
@@ -0,0 +1,523 @@
...
+		Flags: []cli.Flag{
+			cli.StringFlag{
+				Name:  "format",
+				Value: "text",
+				Usage: `The output format for printing the introspection details.
+
+: <format> is a string and must be one of:
+
+    **text**
+    :  Print output in unstructured text suitable for a human to read.
```
[the remark this answers is not in the document]
Comment (maraino) [reply command/crl/inspect.go:80]: Fixed with 4ac6c55

**Review by dopey — APPROVED** [review]
lgtm

**Pull request merged**